TypeCobol/Compiler/CodeModel/SymbolTable.cs

```diff
@@ -818,6 +795,43 @@ public void AddType(TypeDefinition type)
             Add(Types, type);
         }
 
+        public void AddTypeDataDefinition(TypeDefinition data)
+        {
+            //Add type if it doesn't have children
+            if (data.Children.Count == 0)
+            {
+                Add(DataTypeEntries, data);
+            }
+
+            //If type has children, add the children
```

Comment:
This code is easier to read:
```c#
           //Add the children
            AddTypeChildDataDefinition(data);
            
        }

        public void AddTypeChildDataDefinition(DataDefinition data)
        {
            
            foreach (var dataChild in data.Children)
            {
                var childDataDefinition = dataChild as DataDefinition;
                if (childDataDefinition == null)
                {
                    continue;
                }
                Add(DataTypeEntries, childDataDefinition);
                //Continue to add further children
                AddTypeChildDataDefinition(childDataDefinition);
            }
        }
```

Reply:
If I do this, the types with no children will not be added to the DataTypeEntries dictionary (the foreach will not be played if children.count==0).
At the end of the recursivity, the entry will not have any more children, so it will be added to the dictionary.